cognitive/src/test/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIChatCompletionSuite.scala

```diff
@@ -174,6 +176,45 @@ class OpenAIChatCompletionSuite extends TransformerFuzzing[OpenAIChatCompletion]
     testCompletion(customEndpointCompletion, goodDf)
   }
 
+  test("setResponseFormat should set the response format correctly") {
+    completion.setResponseFormat("text")
+    completion.getResponseFormat shouldEqual Map("type" -> "text")
+
+    completion.setResponseFormat("tExT")
+    completion.getResponseFormat shouldEqual Map("type" -> "text")
+
+    completion.setResponseFormat("json")
+    completion.getResponseFormat shouldEqual Map("type" -> "json_object")
+
+    completion.setResponseFormat("JSON")
+    completion.getResponseFormat shouldEqual Map("type" -> "json_object")
+
+    completion.setResponseFormat("json_object")
+    completion.getResponseFormat shouldEqual Map("type" -> "json_object")
+
+    completion.setResponseFormat("Json_ObjeCt")
+    completion.getResponseFormat shouldEqual Map("type" -> "json_object")
+  }
+
+  test("setResponseFormat should throw an exception for invalid response format") {
+    an[IllegalArgumentException] should be thrownBy {
+      completion.setResponseFormat("invalid_format")
+    }
+  }
+
+  test("setResponseFormat with ResponseFormat should set the response format correctly") {
+    completion.setResponseFormat(OpenAIResponseFormat.TEXT)
+    completion.getResponseFormat shouldEqual Map("type" -> "text")
+
+    completion.setResponseFormat(OpenAIResponseFormat.JSON)
+    completion.getResponseFormat shouldEqual Map("type" -> "json_object")
+  }
+
+  test("setResponseFormatCol should set the response format column correctly") {
+    completion.setResponseFormatCol("response_format_col")
+    completion.getResponseFormatCol shouldEqual "response_format_col"
+  }
+
   def testCompletion(completion: OpenAIChatCompletion, df: DataFrame, requiredLength: Int = 10): Unit = {
```

Comment:
can you add a test that calls the service with this new API param set just as a lil e2e